lib/uplinkc/uplink_definitions.h

```diff
@@ -31,6 +31,19 @@ typedef struct Object           { long _handle; } ObjectRef;
 typedef struct Downloader       { long _handle; } DownloaderRef;
 typedef struct Uploader         { long _handle; } UploaderRef;
 typedef struct EncryptionAccess { long _handle; } EncryptionAccessRef;
+typedef struct CaveatPaths      { long _handle; } CaveatPathsRef;
+typedef struct CaveatNonce      { long _handle; } CaveatNonceRef;
+
+typedef struct Caveat {
+    bool disallow_reads;
+    bool disallow_writes;
+    bool disallow_lists;
+    bool disallow_deletes;
+    CaveatPathsRef allowed_paths;
```

Comment:
Having this as a ref feels weird.

Reply:
Maybe this should be `char **allowed_paths; uint32_t allowed_path_count;`?

Reply:
Alternatively we can skip adding allowed_paths for now.

Reply:
If we skip allowed_paths, would that mean that I would not be able to restrict a sub-apikey to specific path? For my use-case I Need that functionality, so I would like to have this included. My Knowledge with Go and C is restricted, though, so I'm not exactly sure how to fix this.

Reply:
`allowed_paths` can be used only with `encryptedPathPrefix` so currently, it will be hard to use it directly. We have different method to overcome this issue `EncryptionAccess.Restrict`. It will take APIKey and given restrictions `Bucket`/`PathPrefix` (PathPrefix is pure not encrypted prefix) and will return restricted APIKey and EncryptionAccess that can be used.
https://github.com/storj/storj/blob/8b07df37f5f4b3fa2d8e6b908e50763321312a4c/lib/uplink/encryption.go#L76

Reply:
Sounds like an alternative - but that is not yet included with uplinkc, too, Right?

Reply:
No, its not. Its also needs to be exported but maybe with this you can avoid many changes (e.g. remove CaveatPaths from PR) in this PR.